src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,40 +1,62 @@
 import React from 'react';
+import { useDispatch } from 'react-redux';
+
+import { useAppSelector } from '../../app/hooks';
+import { filterSlice } from '../../features/filter';
 
 export const TodoFilter: React.FC = () => {
+  const filter = useAppSelector(state => state.filter);
```

Comment:
Please, try to use [createSelector](https://redux-toolkit.js.org/api/createSelector)